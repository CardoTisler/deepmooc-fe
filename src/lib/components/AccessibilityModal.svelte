<script lang="ts">
	import { Section } from '$lib/components';
	import theme from '$lib/shared/stores/theme';
	export let toggleMenu: () => void;

	const enableContrastMode = () => {
		theme.set('contrast');
	};
	const enableLightMode = () => {
		theme.set('light');
	};
	$: isContrastMode = $theme === 'contrast';
</script>

<Section class="w-full max-w-4xl p-4 {isContrastMode ? 'black-bg' : ''}">
	<div class="flex flex-col">
		<div class="flex justify-end">
			<!-- TODO: Replace with X icon -->
			<button class="p-2 focus:border-2 focus:border-blue-600" on:click={toggleMenu}>X</button>
		</div>
		<div>
			<h3 class="dark:text-yellow-300">Juurdepääsetavus</h3>
			<p class="dark:text-yellow-300">
				Veebileht on disainitud põhimõttega, et lehel olev informatsioon ning funktsionaalsused on
				ligipääsetavad võimalikult paljudele inimestele.
			</p>
			<br />
			<!-- <p>Loe rohkem <a href="#" class="underline">juurdepääsetavusest</a></p> -->
		</div>
		<div>
			<h3 class="dark:text-yellow-300">Teksti suurus</h3>
			<p class="dark:text-yellow-300">
				Enimkasutatavad veebibrauserid pakuvad võimalust suurendada või vähendada teksti suurust
				Ctrl nupu (Cmd nupp OS X operatsioonisüsteemi korral) all hoidmise ning + või - vajutamise
				kaudu. Teine võimalus on kasutada arvutihiirt: hoides all Ctrl nuppu ning liigutades
				rullikut samaaegselt. Sätete taastamiseks saab korraga alla vajutada Ctrl ja 0.
			</p>
			<br />
		</div>
		<div>
			<h3 class="dark:text-yellow-300">Kontrast</h3>
			<br />
			<div class="flex flex-row w-full xl:w-1/2 justify-between">
				<button
					class="text-xl border-2 border-primary-200 rounded-xl py-2 px-3 hover:bg-primary-200 dark:text-yellow-300 dark:border-yellow-300 dark:hover:bg-gray-900 focus:border-blue-600 focus:first"
					on:click={enableLightMode} autofocus>Tava sätted</button
				>
				<button
					class="text-xl focus:border-2 bg-gray-800 text-white rounded-xl py-2 px-3 focus:border-blue-600 hover:bg-gray-900 dark:text-yellow-300"
					on:click={enableContrastMode}
					>Must taust ja <span class="text-yellow-300">kollane tekst</span></button
				>
			</div>
		</div>
	</div>
</Section>

<style>
</style>
