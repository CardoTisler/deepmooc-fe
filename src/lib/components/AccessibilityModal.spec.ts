import AccessibilityModal from './AccessibilityModal.svelte';
import { render } from '@testing-library/svelte';
import { describe, expect, it } from 'vitest';

describe('AccessibilityModal.svelte', () => {
	it('should render component', () => {
		const toggleMenu = () => {
			return;
		};
		const { container } = render(AccessibilityModal, { toggleMenu });

		expect(container).toMatchSnapshot();
	});
});
