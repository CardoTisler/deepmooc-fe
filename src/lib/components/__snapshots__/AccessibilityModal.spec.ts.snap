// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`AccessibilityModal.svelte > should render component 1`] = `
<body>
  <div>
    <div
      class="border-2 border-gray-200 rounded-xl bg-white w-full dark:border-yellow-300 w-full max-w-4xl p-4  s-u_aBuH8kPm0b"
    >
      <div
        class="flex flex-col s-dFbO1oczX-LX"
      >
        <div
          class="flex justify-end s-dFbO1oczX-LX"
        >
          <button
            class="p-2 focus:border-2 focus:border-blue-600 dark:text-yellow-300 s-dFbO1oczX-LX"
          >
            X
          </button>
        </div>
         
        <div
          class="s-dFbO1oczX-LX"
        >
          <h3
            class="dark:text-yellow-300 s-dFbO1oczX-LX"
          >
            Juurdepääsetavus
          </h3>
           
          <p
            class="dark:text-yellow-300 s-dFbO1oczX-LX"
          >
            Veebileht on disainitud põhimõttega, et lehel olev informatsioon ning funktsionaalsused on
				ligipääsetavad võimalikult paljudele inimestele.
          </p>
           
          <br
            class="s-dFbO1oczX-LX"
          />
        </div>
         
        <div
          class="s-dFbO1oczX-LX"
        >
          <h3
            class="dark:text-yellow-300 s-dFbO1oczX-LX"
          >
            Teksti suurus
          </h3>
           
          <p
            class="dark:text-yellow-300 s-dFbO1oczX-LX"
          >
            Enimkasutatavad veebibrauserid pakuvad võimalust suurendada või vähendada teksti suurust
				Ctrl nupu (Cmd nupp OS X operatsioonisüsteemi korral) all hoidmise ning + või - vajutamise
				kaudu. Teine võimalus on kasutada arvutihiirt: hoides all Ctrl nuppu ning liigutades
				rullikut samaaegselt. Sätete taastamiseks saab korraga alla vajutada Ctrl ja 0.
          </p>
           
          <br
            class="s-dFbO1oczX-LX"
          />
        </div>
         
        <div
          class="s-dFbO1oczX-LX"
        >
          <h3
            class="dark:text-yellow-300 s-dFbO1oczX-LX"
          >
            Kontrast
          </h3>
           
          <br
            class="s-dFbO1oczX-LX"
          />
           
          <div
            class="flex flex-row w-full xl:w-1/2 justify-between s-dFbO1oczX-LX"
          >
            <button
              autofocus=""
              class="text-xl border-2 border-primary-200 rounded-xl py-2 px-3 hover:bg-primary-200 dark:text-yellow-300 dark:border-yellow-300 dark:hover:bg-gray-900 focus:border-blue-600 focus:first s-dFbO1oczX-LX"
            >
              Tava sätted
            </button>
             
            <button
              class="text-xl focus:border-2 bg-gray-800 text-white rounded-xl py-2 px-3 focus:border-blue-600 hover:bg-gray-900 dark:text-yellow-300 s-dFbO1oczX-LX"
            >
              Must taust ja 
              <span
                class="text-yellow-300 s-dFbO1oczX-LX"
              >
                kollane tekst
              </span>
            </button>
          </div>
        </div>
      </div>
    </div>
    <!--&lt;Section&gt;-->
    <!--&lt;AccessibilityModal&gt;-->
  </div>
</body>
`;
